dipper/sources/PostgreSQLSource.py

```diff
@@ -69,61 +68,53 @@ def fetch_from_pgdb(self, tables, cxn, limit=None, force=False):
                     query = ' '.join((query, "LIMIT", str(limit)))
                     countquery = ' '.join((countquery, "LIMIT", str(limit)))
 
+                cur.execute(countquery)
+                tablerowcount = cur.fetchone()[0]
+
                 outfile = '/'.join((self.rawdir, tab))
 
-                filerowcount = -1
-                tablerowcount = -1
-                if not force:
-                    # check local copy.  assume that if the # rows are the same,
-                    # that the table is the same
-                    # TODO may want to fix this assumption
-                    if os.path.exists(outfile):
-                        # get rows in the file
-                        filerowcount = self.file_len(outfile)
-                        LOG.info(
-                            "(%s) rows in local file for table %s", filerowcount, tab)
-
-                    # get rows in the table
-                    # tablerowcount=cur.rowcount
-                    cur.execute(countquery)
-                    tablerowcount = cur.fetchone()[0]
-
-                # rowcount-1 because there's a header
-                if force or filerowcount < 0 or (filerowcount-1) != tablerowcount:
-                    if force:
-                        LOG.info("Forcing download of %s", tab)
-                    else:
-                        LOG.info(
-                            "%s local (%d) different from remote (%d); fetching.",
-                            tab, filerowcount, tablerowcount)
-                    # download the file
-                    LOG.info("COMMAND:%s", query)
-                    outputquery = """
-    COPY ({0}) TO STDOUT WITH DELIMITER AS '\t' CSV HEADER""".format(query)
-                    with open(outfile, 'w') as f:
-                        cur.copy_expert(outputquery, f)
-                else:
-                    LOG.info("local data same as remote; reusing.")
+                # download the file
+                LOG.info("COMMAND:%s", query)
+                outputquery = """
+                   COPY ({0}) TO STDOUT WITH DELIMITER AS '\t' CSV HEADER""".format(query)
+                with open(outfile, 'w') as f:
+                    cur.copy_expert(outputquery, f)
+
+                filerowcount = self.file_len(outfile)
+
+                if (filerowcount - 1) < tablerowcount:
+                    raise Exception(
+                        "Download from %s failed, %s != %s",
+                        cxn['host'] + ':' + cxn['database'],
+                        (filerowcount - 1),
+                        tablerowcount
+                    )
+                elif (filerowcount - 1) > tablerowcount:
+                    LOG.warning(
+                        "Fetched from %s more rows in file (%s) than reported in count(%s)",
```

Comment:
line's a bit too long (per our endless pylint convo haha)

Reply:
thanks! forgot to lint that file